File: LeetCode/821.cpp
//Link: https://leetcode.com/problems/shortest-distance-to-a-character/

class Solution {
public:
    vector<int> shortestToChar(string s, char c) {
        vector<int> answer(s.length(), INT_MAX);
        int n = s.length();
        for(int i = 0 ; i < n ; i++) {
            if(s[i] == c) {
                for(int j = i ; j >= 0 ; j--) {
                    if(abs(i-j) < answer[j]) answer[j] = abs(i-j);
                    else break;
                }
                for(int j = i + 1 ; j < n ; j++) {
                    if(abs(i-j) < answer[j]) answer[j] = abs(i-j);
                    else break;
                }
            }
        }
        return answer;
    }
};

File: LeetCode/py/821.py
#Link: https://leetcode.com/problems/shortest-distance-to-a-character/

class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        n = len(s)
        answer = [n]*n
        for i in range(n):
            if s[i] == c:
                for j in range(i,-1,-1):
                    if abs(i-j) < answer[j]: answer[j] = abs(i-j)
                    else: break
                for j in range(i+1,n):
                    if abs(i-j) < answer[j]: answer[j] = abs(i-j)
                    else: break
        return answer
